docs/dom-testing-library/cheatsheet.md

```diff
@@ -77,12 +77,18 @@ See [Which query should I use?](guide-which-query.md)
 
 ## Async
 
-See [Async API](api-async.md)
-
-- **wait** retry function within until it stops throwing or times out
-- **waitForElement** retry function or array of functions and return the result
-- `findBy` and `findAllBy` queries are async and retry until either a timeout or
-  if the query returns successfully; they wrap `waitForElement`
+See [Async API](dom-testing-library/api-async.md)
+
+- **wait** (Promise) retry function within until it stops throwing or times out
+- **waitForElement** (Promise) retry function or array of functions and return
```

Comment:
Or array of functions? You mean like `waitForElement([() => {}, () => {}])` I don't think that works does it?

Reply:
Thank you for reviewing! 
I didn't edit that part though but is it saying about following things maybe?

```js
// https://testing-library.com/docs/dom-testing-library/api-async#waitforelement

// return single function
await waitForElement(
  () => getByLabelText(container, 'username'),
  { container }
)

// return array of function
await waitForElement(
  () => [
    getByLabelText(container, 'username'),
    getByLabelText(container, 'password'),
  ],
  { container }
)
```

Is it better to change it simply "retry function and return the result" ?

Reply:
@kentcdodds 
Thank you for reviewing! 
I didn't edit that part though but is it saying about following things maybe?

```js
// https://testing-library.com/docs/dom-testing-library/api-async#waitforelement

// return single function
await waitForElement(
  () => getByLabelText(container, 'username'),
  { container }
)

// return array of function
await waitForElement(
  () => [
    getByLabelText(container, 'username'),
    getByLabelText(container, 'password'),
  ],
  { container }
)
```

Is it better to change it simply "retry function and return the result" ?

Reply:
Weird. Yeah, I'm not sure how that slipped in because that's not possible: https://github.com/testing-library/dom-testing-library/blob/e0eed7dae461cf8f66b28b1e67e9eb5b07bd31cd/src/wait-for-element.js#L18-L23

We should update that. I'll go ahead and fix that now.

Reply:
Oh oh, sorry, I misunderstood! It's not a "function or array of functions" it's "retry the function until it returns an element or an array of elements"

That's what's documented and that's what's supported :)

Reply:
thanks, I fiexed that part.